api/spec/requests/spree/api/orders_controller_spec.rb

```diff
@@ -815,6 +815,38 @@ module Spree
           expect(response.status).to eq 201
           expect(json_response["user_id"]).to eq(user.id)
         end
+
+        context 'with payment', js: true do
+          let(:params) do
+            {
+              payments: [{
+                amount: '10.0',
+                payment_method: payment_method.name,
+                source: {
+                  month: "01",
+                  year: Date.today.year.to_s.last(2),
+                  cc_type: "123",
+                  last_digits: "1111",
+                  name: "Credit Card"
+                }
+              }]
+            }
+          end
+          let!(:payment_method) { create(:payment_method) }
+
+          context "with source" do
+            it "creates a payment" do
+              post spree.api_orders_path, params: { order: params }
+              expect(response.status).to eq 201
+            end
+
+            it "returns an error" do
```

Comment:
I think that this spec would benefit from a context that explains what's happening here, for example
`context "when payment_method is missing" do`

Reply:
done here https://github.com/solidusio/solidus/pull/3066/files#diff-72468e52a59aedfe8a3443597f1a6f3aR836